.circleci/config.yml

```diff
@@ -273,8 +276,61 @@ commands:
           when: always
           path: ~/generated-tests
           destination: generated-tests
+  publish-cloudsmith:
+    steps:
+      - cloudsmith/ensure-api-key
+      - cloudsmith/install-cli
+      - run:
+          name: Publish Packages
+          command: |
+            .circleci/scripts/publish-cloudsmith.sh
 
 workflows:
+  nightly-publish:
```

Comment:
These nightly jobs are petty heavy - these will add 3+ full smoke test and IT runs daily even if there are no changes in the given branches.

Can we find a way to reduce the compute time required?

Reply:
I wanted to make sure we don't publish something unless it actually truly passes everything.  I guess the alternative would be to always publish the first passing package of the day, and keep track of it somewhere or query the repo?

Reply:
Just some naive quick idea, can we add a gate something like here: https://github.com/opennms-forge/opennms-base-container/blob/master/.circleci/build.sh to just build if we have changes in the given branch?

Reply:
OK, latest version adds to the "normal" full test builds, but checks against the last commit date of a previously-published branch.  Stuck a temporary commit at the top of the branch to let it run through everything to make sure it does the right thing, then I'll hit it one more time to make sure it does nothing.  :)